api/capi/src/nntrainer.cpp

```diff
@@ -243,9 +243,9 @@ int ml_nnmodel_destruct(ml_nnmodel_h model) {
 
   if (nnmodel->optimizer)
     delete nnmodel->optimizer;
-  for (auto iter = nnmodel->layers.begin(); iter != nnmodel->layers.end();
-       ++iter)
-    delete (*iter);
+  for (auto &x : nnmodel->layers_map)
```

Comment:
I think nnmodel->layers and nnmodel->layers_map are not same number of layer due to the activation and loss layer. Is it alright if we delete layer according to layers_map?

Reply:
Yes, nnmodel->layers_map will not contain all the layers.
This loop deletes memory allocated corresponding to `struct ml_nnlayer` object.
The internal layers are deleted with finalize (just like you said).